api/v1beta1/azuremanagedcontrolplane_webhook.go

```diff
@@ -434,10 +422,22 @@ func (m *AzureManagedControlPlane) validateManagedClusterNetwork(cli client.Clie
 		if err != nil {
 			allErrs = append(allErrs, field.Invalid(field.NewPath("Cluster", "Spec", "ClusterNetwork", "Services", "CIDRBlocks"), serviceCIDR, fmt.Sprintf("failed to parse cluster service cidr: %v", err)))
 		}
-		ip := net.ParseIP(*m.Spec.DNSServiceIP)
-		if !cidr.Contains(ip) {
+
+		dnsIP := net.ParseIP(*m.Spec.DNSServiceIP)
+		if dnsIP == nil { // dnsIP will be nil if the string is not a valid IP
+			allErrs = append(allErrs, field.Invalid(field.NewPath("Spec", "DNSServiceIP"), *m.Spec.DNSServiceIP, "must be a valid IP address"))
+		}
```

Comment:
Ideally, we would want to return the errors here rather than checking if it is a `.10` IP address. 
I wanted to change it to something like
```suggestion
			allErrs = append(allErrs, field.Invalid(field.NewPath("Spec", "DNSServiceIP"), *m.Spec.DNSServiceIP, "must be a valid IP address"))
			if len(allErrs) > 0 {
				return kerrors.NewAggregate(allErrs.ToAggregate().Errors())
			}
```

Reply:
But if I do this, webhook will skip running `validatePrivateEndpoints()`.

Reply:
What are your thoughts ?